cpp/CMakeLists.txt

```diff
@@ -132,7 +132,20 @@ include(cmake/modules/ConfigureCUDA.cmake)
 ##############################################################################
 # - Requirements -------------------------------------------------------------
 
-if(distance IN_LIST raft_FIND_COMPONENTS OR RAFT_COMPILE_LIBRARIES OR RAFT_COMPILE_DIST_LIBRARY)
+if(RAFT_COMPILE_LIBRARIES)
+  set(RAFT_COMPILE_DIST_LIBRARY ON)
+  set(RAFT_COMPILE_NN_LIBRARY ON)
+endif()
+
+if(distance IN_LIST raft_FIND_COMPONENTS)
+  set(RAFT_COMPILE_DIST_LIBRARY ON)
+endif()
+
+if(nn IN_LIST raft_FIND_COMPONENTS)
```

Comment:
The only change to cpp/CMakeLists.txt in this PR that is actually necessary for scikit-build to work correctly is the change on line 383 to install raft_distance headers, which were otherwise unavailable. The changes on line 135 are just cleanup since `RAFT_COMPILE_LIBRARIES` should imply compiling both distance and nn libraries.

Currently for scikit-build I find the distance component (in `find_package`) if `FIND_RAFT_CPP=ON`, otherwise I build the library with `RAFT_COMPILE_DIST_LIBRARY=ON`. So the question is, what happens if a user has precompiled a raft C++ _without_ compiling the distance library, then sets `FIND_RAFT_CPP=ON` when building the Python library? Do we check that the distance library was compiled (by looking for the appropriate target resulting from `find_package`) and then raise an error?

Reply:
Yes, if the user has raft installed already and want to build the python package without specifying they'd like to build the distance precompiled lib then I'd prefer to throw an error in this case and maybe provide a message to let them know how they can fix it.

Reply:
I've addressed your concern, but I'm not entirely happy with the way that I've done it and I'd like a second look at the way I'm doing it [here](https://github.com/rapidsai/raft/pull/633/files#diff-d6fe2bfef44eb2c5680c98bd4fd14642b7c70627cca040ce94ef0d95240d3b9cR39). I'm not sure why the export set from the C++ library is structured the way that it is. Maybe the way that I've done it is the best that we can do, but it seems like the `find_package` call should just fail here.